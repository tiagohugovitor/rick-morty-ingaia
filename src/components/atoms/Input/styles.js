import styled from 'styled-components';

const StyledInput = styled.input`
  border: 2px solid ${props => props.theme.palette.white.main};
  box-sizing: border-box;
  border-radius: 8px;
  color: ${props => props.theme.palette.text.primary};
  padding: 10px 14px;
  background: transparent;
  font-style: normal;
  font-weight: bold;
  font-size: ${props => props.theme.typography.size.medium};
  line-height: 22px;
`;


export {
  StyledInput,
}
