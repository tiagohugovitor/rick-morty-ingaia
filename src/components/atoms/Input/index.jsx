import React from 'react';
import PropTypes from 'prop-types';
import {StyledInput} from './styles';

const Input = ({
  id,
  type,
  name,
  placeholder,
  onChange,
  value,
  fontSize,
  fontWeight,
}) => (
  <StyledInput
    aria-label={placeholder}
    id={id}
    type={type}
    name={name}
    onChange={(event) => onChange(event.target.value)}
    value={value}
    fontSize={fontSize}
    fontWeight={fontWeight}
    placeholder={placeholder}
  />
);

Input.propTypes = {
  id: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  fontSize: PropTypes.string,
  fontWeight: PropTypes.string,
};

Input.defaultProps = {
  id: 'input-id',
  type: 'text',
  name: 'input-name',
  placeholder: '',
  onChange: () => {},
  value: '',
  fontSize: 'small',
  fontWeight: 'normal',
};

export default Input;
