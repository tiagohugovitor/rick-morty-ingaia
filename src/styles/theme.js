import { createMuiTheme } from '@material-ui/core/styles';

const theme = createMuiTheme({
  typography: {
    fontFamily: "nunito",
    size: {
      mini: '12px',
      tiny: '13px',
      small: '14px',
      mediumSmall: '16px',
      medium: '18px',
      large: '20px',
      huge: '24px',
      extra: '28px'
    },
    weight: {
      light: '300',
      normal: '400',
      bold: '700'
    }
  },
  palette: {
    primary: {
      main: "#CBD736",
      secondary: "#C9D636"
    },
    text: {
      primary: "#9F9F9F",
      secondary: "#8C8C8C",
      tertiary: "#CBD736",
      white: "#FFFFFF",
      black: "#000000",
      grey: "#D3D3D3"
    },
    warning: {
      main: "#FF0000"
    },
    white: {
      main: "#FFFFFF"
    },
    black: {
      main: "#000000"
    },
    background:{
      primary: "#1E1E1E",
      secondary: "#303030",
      tertiary: "#1B1B1B"
    },
    border: {
      primary: "#606060"
    }
  }
});

export default theme;