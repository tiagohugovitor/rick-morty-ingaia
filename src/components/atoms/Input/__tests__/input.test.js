import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import { ThemeProvider } from 'styled-components'
import theme from '../../../../styles/theme';
import Input from '../index';

configure({ adapter: new Adapter() });


describe('Render Input Component', () => {
  it('render correctly input component', () => {
    const InputComponent = renderer.create(
      <ThemeProvider theme={theme}>
        <Input />
      </ThemeProvider>
    ).toJSON();
    expect(InputComponent).toMatchSnapshot();
  });

  it('render correctly with default value of props', () => {
    const defaultProps = {};
    const InputComponent = mount(
      <ThemeProvider theme={theme}>
        <Input {...defaultProps} />
      </ThemeProvider>
    ).find(Input).first();
    expect((InputComponent).prop('id')).toEqual('input-id');
    expect((InputComponent).prop('type')).toEqual('text');
    expect((InputComponent).prop('name')).toEqual('input-name');
    expect((InputComponent).prop('placeholder')).toEqual('');
    expect((InputComponent).prop('value')).toEqual('');
    expect((InputComponent).prop('fontSize')).toEqual('small');
    expect((InputComponent).prop('fontWeight')).toEqual('normal');
  });

  it('render correctly with null values of props', () => {
    const nullProps = {
      id: null,
      type: null,
      name: null,
      placeholder: null,
      onChange: null,
      value: null,
      fontSize: null,
      fontWeight: null,
    };
    const InputComponent = mount(
      <ThemeProvider theme={theme}>
        <Input {...nullProps} />
      </ThemeProvider>
    ).find(Input).first();
    expect((InputComponent).prop('id')).toEqual(null);
    expect((InputComponent).prop('type')).toEqual(null);
    expect((InputComponent).prop('name')).toEqual(null);
    expect((InputComponent).prop('placeholder')).toEqual(null);
    expect((InputComponent).prop('value')).toEqual(null);
    expect((InputComponent).prop('fontSize')).toEqual(null);
    expect((InputComponent).prop('fontWeight')).toEqual(null);
  });

  it('render correctly with normal value of props', () => {
    const props = {
      id: "search-bar-input",
      type: "text",
      name: "search-characters-input",
      placeholder: "Search characters",
      value: 'tests',
      fontSize: "mediumSmall",
      fontWeight: "bold",
    };
    const InputComponent = mount(
      <ThemeProvider theme={theme}>
        <Input {...props} />
      </ThemeProvider>
    ).find(Input).first();
    expect((InputComponent).prop('id')).toEqual("search-bar-input");
    expect((InputComponent).prop('type')).toEqual("text");
    expect((InputComponent).prop('name')).toEqual("search-characters-input");
    expect((InputComponent).prop('placeholder')).toEqual("Search characters");
    expect((InputComponent).prop('value')).toEqual("tests");
    expect((InputComponent).prop('fontSize')).toEqual("mediumSmall");
    expect((InputComponent).prop('fontWeight')).toEqual("bold");
  });

  it('check onChange callback', () => {
    const onChange = jest.fn();
    const props = {
      onChange: onChange,
    };
    const InputComponent = mount(
      <ThemeProvider theme={theme}>
        <Input {...props} />
      </ThemeProvider>
    ).find(Input).first();
    InputComponent.simulate('change', { target: { value: 'Onchange Test' } });
    expect(onChange).toHaveBeenCalledWith('Onchange Test');
  });

});