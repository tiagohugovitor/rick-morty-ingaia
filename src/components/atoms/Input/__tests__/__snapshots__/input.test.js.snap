// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Input Component render correctly input component 1`] = `
<input
  aria-label=""
  className="sc-bdvvaa fPngCV"
  fontSize="small"
  fontWeight="normal"
  id="input-id"
  name="input-name"
  onChange={[Function]}
  placeholder=""
  type="text"
  value=""
/>
`;
